app/Listeners/WebhookReceivedListener.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Listeners;
+
+use App\Models\Plan;
+use App\Models\Subscription;
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+use Laravel\Paddle\Customer;
+use Laravel\Paddle\Events\SubscriptionCreated;
+use Laravel\Paddle\Events\WebhookReceived;
+use Laravel\Paddle\Exceptions\InvalidPassthroughPayload;
+use function Safe\json_decode;
+
+class WebhookReceivedListener
+{
+    /**
+     * Handle Paddle webhook event.
+     *
+     * @param  WebhookReceived  $event
+     * @return void
+     */
+    public function handle(WebhookReceived $event)
+    {
+        Log::info("{$event->payload['alert_name']} paddle webhook received", [
+            'payload' => $event->payload,
+        ]);
+
+        $method = 'handle'.Str::studly($event->payload['alert_name']);
+
+        if (method_exists($this, $method)) {
+            try {
+                $this->{$method}($event->payload);
+            } catch (InvalidPassthroughPayload) {
+                // Ignored
+            }
+        }
+    }
+
+    /**
+     * Handle subscription created.
+     *
+     * @param  array  $payload
+     * @return void
+     *
+     * @throws \Laravel\Paddle\Exceptions\InvalidPassthroughPayload
+     */
+    protected function handleSubscriptionCreated(array $payload)
+    {
+        $customer = $this->findOrCreateCustomer($payload, $payload['passthrough']);
+
+        $trialEndsAt = $payload['status'] === Subscription::STATUS_TRIALING
+            ? Carbon::createFromFormat('Y-m-d', $payload['next_bill_date'], 'UTC')->startOfDay()
```

Comment:
By default it's UTC and by default it's also start of day, but at least we are sure.

Reply:
See https://github.com/laravel/cashier-paddle/blob/198b40a790acad4a1b0ace3f6dab3794bac0315c/src/Http/Controllers/WebhookController.php#L233